core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/multitenancy/MultitenantMessageReceiver.java

```diff
@@ -436,6 +436,7 @@ private void doServletRest(MessageContext mainInMsgContext, String to,
 	    mainOutMsgContext.setProperty(Constants.Configuration.MESSAGE_TYPE,
                   tenantOutMsgContext.getProperty(Constants.Configuration.MESSAGE_TYPE));
             AxisEngine.send(mainOutMsgContext);
```

Comment:
@ajanthan sure will add a try catch and finally block, will add this code into finally block.